src/app/App.tsx

```diff
@@ -89,33 +60,30 @@ const App: React.ComponentType = () => {
             endTime={ endTime }
             timeZone={ timezone }
         />;
-    }
+    }}
 
     return (
         <AppContext.Provider value={ {
             rbac,
             server,
             isOrgAdmin: !!isOrgAdmin
         } }>
-            { hasReadPermissions ? (
-                <>
-                    <NotificationsPortal />
-                    <InsightsEnvDetector insights={ insights } onEnvironment={ staging }>
-                        <RenderIfTrue>
-                            <Switch
-                                className={ switchClassname }
-                                isChecked={ usingExperimental }
-                                onChange={ toggleExperimental }
-                                labelOff="Enable experimental features"
-                                label="Disable experimental features"
-                            />
-                        </RenderIfTrue>
-                    </InsightsEnvDetector>
-                    <Routes />
-                </>
-            ) : (
-                <NotAuthorized serviceName={ serviceName } />
-            ) }
+            <>
+                <NotificationsPortal />
+                <InsightsEnvDetector insights={ insights } onEnvironment={ staging }>
+                    <RenderIfTrue>
+                        <Switch
+                            className={ switchClassname }
+                            isChecked={ usingExperimental }
+                            onChange={ toggleExperimental }
+                            labelOff="Enable experimental features"
+                            label="Disable experimental features"
+                        />
+                    </RenderIfTrue>
+                </InsightsEnvDetector>
+                <Routes />
+            </>
```

Comment:
Wrapping this all in the `CheckReadPermissions` 
`CheckReadPermissions` will decide if it shows the content or not

```suggestion
            <CheckReadPermissions>
                <NotificationsPortal />
                <InsightsEnvDetector insights={ insights } onEnvironment={ staging }>
                    <RenderIfTrue>
                        <Switch
                            className={ switchClassname }
                            isChecked={ usingExperimental }
                            onChange={ toggleExperimental }
                            labelOff="Enable experimental features"
                            label="Disable experimental features"
                        />
                    </RenderIfTrue>
                </InsightsEnvDetector>
                <Routes />
            <CheckReadPermissions />
```